crates/core/app/tests/app_tracks_uptime_for_validators_only_once_active.rs

```diff
@@ -73,19 +70,12 @@ async fn app_tracks_uptime_for_validators_only_once_active() -> anyhow::Result<(
     };
 
     // Get the identity key of the genesis validator, before we go further.
-    // Retrieve the validator definition from the latest snapshot.
-    let existing_validator_id = {
-        use penumbra_stake::component::validator_handler::validator_store::ValidatorDataRead;
-        let validators = &storage
-            .latest_snapshot()
-            .validator_definitions()
-            .tap(|_| info!("getting validator definitions"))
-            .await?;
-        match validators.as_slice() {
-            [Validator { identity_key, .. }] => *identity_key,
-            unexpected => panic!("there should be one validator, got: {unexpected:?}"),
-        }
-    };
+    let [existing_validator_id] = storage
+        .latest_snapshot()
+        .validator_identity_keys()
+        .await?
+        .try_into()
+        .map_err(|keys| anyhow::anyhow!("expected one key, got: {keys:?}"))?;
```

Comment:
an example of the benefit of being able to request a `Vec<IdentityKey>` out of the temporary storage's latest snapshot.